lib/rake/testtask.rb

```diff
@@ -128,7 +128,7 @@ def ruby_opts_string # :nodoc:
       opts = @ruby_opts.dup
       opts.unshift("-I\"#{lib_path}\"") unless @libs.empty?
       opts.unshift("-w") if @warning
-      opts.unshift('--verbose') if @verbose
+      opts.unshift('-W2') if @verbose
```

Comment:
might this conflict with the `-W` option passed to Rake to show task source files?